### DISCUSSION TITLE
Add indent and dedent in insert mode

### DISCUSSION THREAD

**Discussion by pickfire** [helix-editor/helix#10387] [Ideas]
Vim have this useful feature to allow indent and dedent, kakoune does not have this which I missed.

I think it will be useful at this stage when we don't have auto-indent or auto-dedent. Rather than `<backspace><backspace><backspace><backspace>` we can do `ctrl-d` once in insert mode to cancel out 4 spaces indented.

It will probably use similar mechanism to `<` and `>` but I don't know how it will affect `.` yet.

Planning to do it later when I have time, probably after infobox.

**Comment by CptPotato** [comment]
I've seen other editors use shift+tab for that purpose. Not sure if this is in line with modal editors though.

**Comment by sudormrfbin** [comment]
`ctrl-t` and `ctrl-d` in vim indents and dedents even if the cursor is on the middle or end of the line. So if we go with `shift-tab` we wouldn't have a counterpart. How about `ctrl-,` and `ctrl-.` (same keys as `<` and `>`) ?

**Comment by pickfire** [comment]
> How about ctrl-, and ctrl-. (same keys as < and >) ?

That works but not discoverable. Who would have thought that it is the version of < and > without shift? And that is only assuming that the users use qwerty.

shift-tab is hard probably because it's in a terminal, so we shift-tab to both indent and dedent?

**Comment by sudormrfbin** [comment]
Ah the qwerty point is good enough to drop those suggestions. Maybe we should just go with vim's defaults (also right now we strip `shift` from all key events).

**Comment by TornaxO7** [comment]
Can this issue be closed or is the suggestion here to have a default keybinding for this?
